### DISCUSSION TITLE
Using private RPC ( GasHawk ) for official Era bridge , expiration period?

### DISCUSSION THREAD

**Discussion by qnfr** [zkSync-Community-Hub/zksync-developers#219] [General] [answered]
### Environment

Mainnet

### Provide a brief description of the functionality you're trying to implement.

For anyone not familiar, GasHawk is a RPC , that queues your transaction in ETH Mainnet, and when the gas prices are cheap, it executes the tx, you can set a deadline e.t.c. I was wondering if official bridge contract has any expiration period, I mean you can make GasHawk wait maximum to 24h and search for cheaper gas. Meanwhile does the bridge waits? or it Reverts?

### What is the specific issue or error you're encountering?

.

### Can you share the error messages or logs you're receiving, if any?

.

### Have you made any recent changes to the contract before encountering this issue?

.

### Are there any external libraries or contracts that your contract interacts with?

.

### Can you provide the relevant portions of your contract code where the issue is occurring?

.

### Have you tried to isolate the problem, and if so, what were the results?

.

### What steps have you already taken to try to resolve the issue?

.

**Comment by MexicanAce** [comment, marked as answer]
Hi @qnfr 👋 

Thanks for the inquiry, it sounds like this functionality is specific to GasHawk and not controlled by the zkSync network. For more details on how GasHawk works, I would reach out to their docs for more information.

**Reply by qnfr** [reply]
Dear Ace,
I wrote to them first already. They don't have any experience with Layer 2's. So they adviced me to contact zkSync Support. The most important question is do the official era bridge have tx expiration period. Because Gashawk takes the tx and put its on hold and reacts it when gwei is low. You can set period up to 24h.

**Reply by MexicanAce** [reply]
I don't believe that's native functionality within zkSync Era Bridge

**Comment by discpenromo** [comment]
GasHawk's functionality involves queuing transactions in the Ethereum Mainnet and executing them when gas prices are favorable. However, GasHawk's waiting period and the official bridge contract's expiration period are independent of each other. The official bridge contract does not wait for GasHawk's transactions to execute. Instead, it processes transactions according to its own logic and timeframe, which may or may not align with GasHawk's waiting period. Therefore, GasHawk's waiting period does not affect the behavior of the official bridge contract, and transactions queued by GasHawk may or may not be processed by the bridge within its expiration period.

**Comment by maggielyon** [comment]
GasHawk operates by queuing transactions and executing them when gas prices are low. However, it's important to note that the behavior of the official bridge contract in response to delayed transactions is determined by its implementation. It's recommended to review the bridge contract's source code or documentation to understand how it handles transactions with extended wait times. If there's no explicit mention of expiration periods or timeout mechanisms, further investigation may be required to determine the contract's behavior in such scenarios.

**Comment by educatedb** [comment]
The official bridge contract does not have a built-in expiration period. Transactions sent to the bridge will remain pending until they are processed, regardless of any delay in gas price optimization tools like GasHawk.

**Comment by thakurmatt** [comment]
The user is inquiring about the expiration period of the official bridge contract on Mainnet and whether it will revert or wait if GasHawk queues the transaction for up to 24 hours to search for cheaper gas. No specific issue or error message is provided, and no recent changes to the contract or external libraries/contracts are mentioned. The user has not provided any relevant portions of their contract code or steps taken to resolve the issue. No repository link or additional details are provided.
